### ISSUE TITLE
QAM: Some of the Reminders are Combining, and some Aren't

### ISSUE THREAD

**Issue by michiQ** [yadaguru/yadaguru-app#157]

**Comment by michiQ** [comment]
I think this one might already be in the works on another card, but just wanted to bring it to the board in case in wasn't.  Basically this that when you have schools that have reminders on the same day, some of them combine and some don't.  For example, in this screen shot we can see that Ask Teachers for Recommendations combines Penn and Harvard's reminder, but then Register for Tests isn't combined.

**Comment by michiQ** [comment]
![img_1285](https://cloud.githubusercontent.com/assets/8203279/23590065/592852a4-01a7-11e7-867c-ae5a78e94d81.jpg)
![img_1284](https://cloud.githubusercontent.com/assets/8203279/23590066/5ae1497a-01a7-11e7-9ea0-caff4a146ee8.jpg)

**Comment by bobderrico80** [comment]
I think I know what is going on here. If the due dates were originally on the same day (which looks like the case for the Penn and Harvard example, they will be combined). I am guessing that the duplicates that are showing up in ASAP are doing so because their original due dates are different (so they wouldn't normally be combined), but since all late reminders show up in this list, we see multiples.

We will have to work the logic around deduping to account for this.

**Comment by michiQ** [comment]
I'll also see what I've got set up with asap messages this week.

**Comment by michiQ** [comment]
I'm noticing in test that the reminders that are getting combined only have one time frame.  For example, Register for Tests, Start Essays, and Ask Counselor for Recommendation only have one timeframe and they are the ones that are combining.   I'm going to go ahead and breakout all the reminders that are multiple timeframes into individual reminders for each timeframe to see if that takes care of the issue.

**Comment by michiQ** [comment]
should have them all separated into sepearte reminders tomorrow and I'll test again and see how it turns out.

**Comment by michiQ** [comment]
ok - so I hope I haven't broken the app.  I separated out all the reminders.  Then created two schools with the same date to see if they would combine. and now when i click on the school I get the 'our bad' message.   I tried on both devices that I have and same thing.  I'm currently blocked on test now.

**Comment by bobderrico80** [comment]
Checked the logs, and I do see 500 errors, but don't have a reason why yet. I think the next priority will need to be to complete #124, so we have context around these error (all I know is that you were getting 500 errors, but we don't know what on the server was failing).

FWIW, I did restart the API container. While I doubt that will fix it, it is worth a shot.

I will add some logging statements around that area tonight and redeploy to try to figure out what's up.

**Comment by michiQ** [comment]
Thanks man!  Just checked and still getting the error.  wonder if the app is looking for reminders to group together and isn't finding any and is like 'ohmygawwddidontknowwhattodo'

**Comment by bobderrico80** [comment]
Tracked it down. Not related to grouping, but to subbing out variables on empty late message/details. Fix has been deployed to both environments. You should be able to QA this one now.

**Comment by michiQ** [comment]
looks beautiful - looks great! Thanks!
